src/main/java/buildcraft/factory/BlockTank.java

```diff
@@ -88,123 +88,232 @@ public IIcon getIconAbsolute(IBlockAccess iblockaccess, int i, int j, int k, int
+        // Regular Fluid Container (FC) Contingencies (IC2 / GT / Flask / Bucket)
+        if (FluidContainerRegistry.isContainer(held)) {
+            FluidStack fromItem = FluidContainerRegistry.getFluidForFilledItem(held);
 
-                        // Handle empty containers
+            // container -) tank logic
+            if (fromItem != null) {
+                int moved = clicked.fill(ForgeDirection.UNKNOWN, fromItem, true);
+
+                if (moved > 0 && !BuildCraftCore.debugWorldgen && !creative) {
+                    if (held.stackSize > 1) {
+                        ItemStack emptied = FluidContainerRegistry.drainFluidContainer(held);
+                        addOrDrop(player, emptied);
+                        player.inventory
+                                .setInventorySlotContents(player.inventory.currentItem, InvUtils.consumeItem(held));
                     } else {
-                        FluidStack available = tank.getTankInfo(ForgeDirection.UNKNOWN)[0].fluid;
-
-                        if (available != null) {
-                            ItemStack filled = FluidContainerRegistry.fillFluidContainer(available, current);
-
-                            liquid = FluidContainerRegistry.getFluidForFilledItem(filled);
-
-                            if (liquid != null) {
-                                if (!BuildCraftCore.debugWorldgen && !entityplayer.capabilities.isCreativeMode) {
-                                    if (current.stackSize > 1) {
-                                        if (!entityplayer.inventory.addItemStackToInventory(filled)) {
-                                            return false;
-                                        } else {
-                                            entityplayer.inventory.setInventorySlotContents(
-                                                    entityplayer.inventory.currentItem,
-                                                    InvUtils.consumeItem(current));
-                                        }
-                                    } else {
-                                        entityplayer.inventory.setInventorySlotContents(
-                                                entityplayer.inventory.currentItem,
-                                                InvUtils.consumeItem(current));
-                                        entityplayer.inventory
-                                                .setInventorySlotContents(entityplayer.inventory.currentItem, filled);
-                                    }
-                                }
-
-                                tank.drain(ForgeDirection.UNKNOWN, liquid.amount, true);
-
-                                return true;
-                            }
-                        }
+                        player.inventory.setInventorySlotContents(
+                                player.inventory.currentItem,
+                                FluidContainerRegistry.drainFluidContainer(held));
```

Comment:
you can extract this variable outside the current condition and reuse it in both branches